packages/router-generator/tests/generator.test.ts

```diff
@@ -0,0 +1,90 @@
+import fs from 'fs/promises'
+import { describe, it, expect } from 'vitest'
+
+import { generator, getConfig, type Config } from '../src'
+
+function getFolderDir(folder: string) {
+  return process.cwd() + `/tests/${folder}`
+}
+
+async function setupConfig(folder: string) {
+  const dir = getFolderDir(folder)
+  const config = await getConfig({
+    disableLogging: true,
+    routesDirectory: dir + '/routes',
+    generatedRouteTree: dir + '/routeTree.gen.ts',
+  })
+  return config
+}
+
+async function getRouteTreeFileText(config: Config) {
+  const location = config.generatedRouteTree
+  const text = await fs.readFile(location, 'utf-8')
+  return text
+}
+
+async function getExpectedRouteTreeFileText(folder: string) {
+  const dir = getFolderDir(folder)
+  const location = dir + '/routeTree.expected.ts'
+  const text = await fs.readFile(location, 'utf-8')
+  return text
+}
+
+// Write a test
+describe('with a default config', () => {
+  it('should wire-up the routes for a "single-level" tree', async () => {
+    const folderName = 'single-level'
+    const config = await setupConfig(folderName)
+
+    await generator(config)
+
+    const [expectedRouteTree, generatedRouteTree] = await Promise.all([
```

Comment:
can we reduce the verbosity one more step?

```tsx
describe('with a default config', () => {
it.each([
  ['single-level],
  ['flat'],
  // etc
])(''should wire-up the routes for a "%s" tree', async (folderName) => {
    const config = await setupConfig(folderName)
    await generator(config)
    // ...
});
}
```

OR
we could iterate over all folders instead of enumerating them in the test?

Reply:
I'd go for the first option since this is currently just testing the `generator` function and this could change should we decide to test the `config` related stuff as well.

Also, didn't know that `it.each` was a thing...
PS: Literally my first time ever writing tests... Pretty much yolo'd till now.

Reply:
I'll dump all the dirs inside a folder and get the foldernames from there.